third_party/move/move-model/src/ty.rs

```diff
@@ -2424,44 +2424,42 @@ impl Substitution {
                     .map_err(TypeUnificationError::lift(order, t1, t2))?,
                 ));
             },
-            (Type::Fun(a1, r1, abilities1), Type::Fun(a2, r2, abilities2)) => {
+            (Type::Fun(a1, r1, abilities1), Type::Fun(a2, r2, abilities2))
```

Comment:
Could we add the following test case to show that vectors are invariant (when they contain function types with reference parameters)?

```
//# publish
module 0x66::zz {
    fun apply(fv: vector<|&mut u64| has copy + drop>) {
        let x = 0;
        fv[0](&mut x);
    }

    fun test() {
        let func: |&u64| has copy + drop = |x| { *x + 1; };
        apply(vector[func]);
    }
}
```

It currently fails typecheck (as expected for vector invariance).

I filed a related bug https://github.com/aptos-labs/aptos-core/issues/16403, (likely not to be addressed as part of this PR, but linking here due to the relation).

Reply:
Have a fix in the next PR, also adding the additional example. That became a bit larger as expected, so separated.